### PR TITLE
[FIX] web: make draggable hook tolerant with mouse movements

### PR DESCRIPTION
**Pull request by Iucapad**
Before this commit, when a user moved its mouse from only a few pixels, a kanban record could not be clicked on it, or an item in a sortable list couldn't handle the click, since the dragging state was immediatly applied to the element.

Now, there is a 10 pixels tolerance, meaning that the dragging sequence is only applied once the user has moved the cursor from at least 10px horizontally, or vertically from the original point where the mouse button was pressed and remains down. This reduces frustation when the user didn't intended to move the kanban card with the mouse moving by mistake.

task-3233698

### CODE REVIEW

**Comment by Iucapad** [comment]
I'm annoyed by the testing of this fix. Tests helpers have been improved in master and I should basically copy/paste the master helper for my needs

**Review by aab-odoo — APPROVED** [review]
It's looking good, but I would like @Arcasias to review as well since he's the expert :-)

Thanks @Iucapad

File: addons/web/static/src/core/utils/draggable_hook_builder.js [hunk 1]
```diff
@@ -478,6 +493,22 @@ export function makeDraggableHook(hookParams = {}) {
                 ctx.mouse.y = ev.clientY;
             };
 
+            /**
+             * Returns whether the user has moved from at least the number of pixels
+             * that are tolerated from the initial mouse position.
+             * @param {MouseEvent} ev
+             */
+            const canDrag = (ev) => {
```
[the remark this answers is not in the document]
Comment (Iucapad) [reply addons/web/static/src/core/utils/draggable_hook_builder.js]: Ok then, it should be good now :)
Yes, I expect to have some conflicts in the forward port. Mostly in the utils, where my change is no longer needed

File: addons/web/static/src/core/utils/draggable_hook_builder.js [hunk 2]
```diff
@@ -274,6 +275,19 @@ export function makeDraggableHook(hookParams = {}) {
                 execBuildHandler("onDragStart");
             };
 
+            /**
+             * Returns whether the user has moved from at least the number of pixels
+             * that are tolerated from the initial mouse position.
+             * @param {MouseEvent} ev
+             */
+            const canDrag = (ev) => {
```
[the remark this answers is not in the document]
Comment (Iucapad) [reply addons/web/static/src/core/utils/draggable_hook_builder.js]: f..... I only looked for dragEnd to set it before. Done!

**Pull request closed without merging**